bin/guard

```diff
@@ -2,31 +2,114 @@
 
 require "pathname"
 
-default = Pathname(ENV.fetch("BUNDLE_GEMFILE", "Gemfile"))
-relative_to_binstub = Pathname(__FILE__).realpath + "../../Gemfile"
-gemfile = [default, relative_to_binstub].detect(&:exist?)
-
-if gemfile && !ENV["RUBYGEMS_GEMDEPS"]
-  ENV["BUNDLE_GEMFILE"] = gemfile.to_s
-  require "rubygems"
-  require "bundler/setup"
-else
-  require "rubygems"
-end
+class GuardReloader
+  class Config
+    def using_rubygems?
+      ENV["RUBYGEMS_GEMDEPS"]
+    end
+
+    def setup_rubygems_for_deps
+      require "rubygems"
+    end
+
+    def current_bundler_gemfile
+      ENV["BUNDLE_GEMFILE"]
+    end
+
+    def using_bundler?
```

Comment:
Yes - the variable can be passed through a parent process, and ideally `RUBYOPT` should be set to automatically include bundler/setup in this case. This is to err on the side of caution - the code should end up as a no-op if everything is ok. (Otherwise, users should explicitly use `Bundler.with_clean_env` instead to disable bundler in child processes). The overhead of the code is almost non-existent, while the potential debugging costs otherwise are absolutely huge.